File: src/main/java/com/examportal/dto/QuestionDTO.java
package com.examportal.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class QuestionDTO {
    
    @NotNull(message = "Exam ID must be provided")
    private Long examId;
    
    @NotBlank(message = "Question text must be provided")
    private String questionText;
    
    @NotBlank(message = "Option A must be provided")
    private String optionA;
    
    @NotBlank(message = "Option B must be provided")
    private String optionB;
    
    @NotBlank(message = "Option C must be provided")
    private String optionC;
    
    @NotBlank(message = "Option D must be provided")
    private String optionD;
    
    @NotBlank(message = "Correct option must be provided")
    @Pattern(regexp = "[ABCD]", message = "Correct option must be A, B, C, or D")
    private String correctOption;
    
    @NotNull(message = "Marks must be provided")
    @Positive(message = "Marks must be positive")
    private Integer marks;
}

File: Dotnet3/ExamPortal/DTOs/QuestionDTO.cs
using System.ComponentModel.DataAnnotations;

namespace ExamPortal.DTOs
{
    public class QuestionDTO
    {
        [Required(ErrorMessage = "Exam ID must be provided")]
        public long ExamId { get; set; }
        
        [Required(ErrorMessage = "Question text must be provided")]
        public string QuestionText { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Option A must be provided")]
        public string OptionA { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Option B must be provided")]
        public string OptionB { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Option C must be provided")]
        public string OptionC { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Option D must be provided")]
        public string OptionD { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Correct option must be provided")]
        [RegularExpression("[ABCD]", ErrorMessage = "Correct option must be A, B, C, or D")]
        public string CorrectOption { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Marks must be provided")]
        [Range(1, int.MaxValue, ErrorMessage = "Marks must be positive")]
        public int Marks { get; set; }
    }
}
